File: 3834-minimum-operations-to-convert-all-elements-to-zero/3834-minimum-operations-to-convert-all-elements-to-zero.cpp
#pragma GCC optimize("O3", "unroll-loops");
static const int __ = []() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    return 0;
}();
class Solution {
   public:
    int minOperations(vector<int>& nums) {
        vector<int> stack;
        int ans = 0;
        for (int num : nums) {
            while (!stack.empty() && stack.back() > num) {
                stack.pop_back();
            }

            if (num == 0) continue;
            if (stack.empty() || stack.back() < num) {
                ans++;
                stack.emplace_back(num);
            }
        }

        return ans;
    }
};

File: 3834-minimum-operations-to-convert-all-elements-to-zero/3834-minimum-operations-to-convert-all-elements-to-zero.go
func minOperations(nums []int) int {
	stack := make([]int, 0, len(nums)/2)
	ans := 0

	for _, num := range nums {
		for len(stack) > 0 && stack[len(stack)-1] > num {
			stack = stack[0 : len(stack)-1]
		}

		if num == 0 {
			continue
		}

		if len(stack) == 0 || stack[len(stack)-1] < num {
			ans++
			stack = append(stack, num)
		}
	}

	return ans

}
